### PR TITLE
Convert offer to separate db fields on applicationchoice table

### PR DESCRIPTION
**Pull request by RobertM905**
### Context

Currently Offers are represented in Application Choice as a Json field.
It has been decided that splitting offer conditions into their own string array field.

### Changes proposed in this pull request

- Add `offer_conditions` to Application Choice
- Update `make_an_offer` service and controller to use `offer_conditions`
- Update `make_an_offer` service specs to test new field
- Remove `offer` from ApplicationChoice

### Guidance to review

- Run specs
- Check migrations and db schema to ensure fields are correct.

### Link to Trello card

[1085 - Convert offer to separate db fields on application choice table](https://trello.com/c/PnFLwRMN/1085-convert-offer-to-separate-db-fields-on-applicationchoice-table)

### CODE REVIEW

File: app/services/vendor_api/single_application_presenter.rb [hunk 1]
```diff
@@ -61,5 +61,11 @@ def get_rejection
         }
       end
     end
+
+    def get_offer
```
Comment (tijmenb) [inline app/services/vendor_api/single_application_presenter.rb]: Should this still return `nil` if there's no offer?

Comment (RobertM905) [reply app/services/vendor_api/single_application_presenter.rb]: Yes. Would have to check the ApplicationChoice status as you can still have an offer without conditions :/

Comment (RobertM905) [reply app/services/vendor_api/single_application_presenter.rb]: Updated here: d59a1ff049952e71b2694bcc6c28b08d5dfdab1c
What do you think?

Comment (duncanjbrown) [reply app/services/vendor_api/single_application_presenter.rb]: What about when the ApplicationChoice is in a state like `conditions_met` — the offer should still exist?

Comment (tijmenb) [reply app/services/vendor_api/single_application_presenter.rb]: Ooh good catch @duncanjbrown.

**Pull request closed without merging**